src/nvim/memory.h

```diff
@@ -45,7 +45,16 @@ typedef struct consumed_blk {
   struct consumed_blk *prev;
 } *ArenaMem;
 
-#define ARENA_ALIGN sizeof(void *)
+#ifdef __GNUC__
+# if defined(__i386__)
+#  define ARENA_ALIGN 16
+# else
+#  define ARENA_ALIGN ((2 * sizeof(size_t)) < \
+                       __alignof__(long double) ? __alignof__(long double) : (2 * sizeof(size_t)))
```

Comment:
Is there a source that the recommended alignment on x86 is 16 and not 8? and that the required one on AARCH64 is 16 and not 8? (it can be assumed we never use weird or obsolete x87 doubles, nor 128 bit ints). just hardcoding `ARENA_ALIGN` to `MAX(sizeof(void *), sizeof(double))` would have be been simpler.

Also `__GNUC__`  looks fishy. we only care what the CPU wants, not about gcc/glibc's ideas.

Reply:
Above logic is based purely on:
https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/generic/malloc-alignment.h;h=ef0f381f772088d3e8a8648410bba85b1a970451;hb=c804cd1c00adde061ca51711f63068c103e94eef
https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/generic/malloc-alignment.h;h=ef0f381f772088d3e8a8648410bba85b1a970451;hb=c804cd1c00adde061ca51711f63068c103e94eef

> just hardcoding `ARENA_ALIGN` to `MAX(sizeof(void *), sizeof(double))` would have be been simpler.

I don't really know what are neovim's alignment requirements hence PR attempts to be as generic as possible. If you're sure it will never reach maximum then I guess it's fine.

> Also `__GNUC__` looks fishy. we only care what the CPU wants, not about gcc/glibc's ideas.

`__GNUC__` is there to make sure `__i386__` and `__alignof__` are available.

Reply:
I think, we should try to just do `MAX(sizeof(double), sizeof(void *))` here without any checks. the only issue I can see is that we use 4-byte alignment on ARMv7 where 8-byte is needed, due to 64-bit floats. if there is still an issue even with that people can report a new issue.